src/string.rs

```diff
@@ -110,6 +112,22 @@ impl<'gc> String<'gc> {
             String::Static(b) => b,
         }
     }
+
+    pub fn len(&self) -> Result<i64, StringError> {
+        fn as_i64(len: usize) -> Result<i64, StringError> {
+            if len <= std::i64::MAX as usize {
+                Ok(len as i64)
+            } else {
+                Err(StringError::TooLong)
+            }
+        }
```

Comment:
You should probably use num_traits::cast here, as it's what the rest of the crate does.

Reply:
Ehhh, actually I take that back, probably this is an impossible error, so it should either be a panic or... just return i64::MAX or something.  Probably panic?  I mean how realistic is a string > i64::MAX?.

Reply:
True, unlikely. I'll just panic.